odl/deform/linearized.py

```diff
@@ -231,7 +230,9 @@ def interp_byaxis(self):
     @property
     def interp(self):
         """Interpolation scheme or tuple of per-axis interpolation schemes."""
-        if _all_interp_equal(self.interp_byaxis):
+        if len(self.interp_byaxis) == 0:
```

Comment:
If we didn't have this special case, then it would fall under the next one (because `all(empty collection)` evaluates to `True`), and then fail in the indexing operation.
But maybe it's simpler to just add a `len(self.interp_byaxis) != 0 and` to that condition and handle the empty case in the `else`.